pkg/instance/service.go

```diff
@@ -184,8 +188,55 @@ func (s service) unlink(id uint) error {
 	return s.instanceRepository.Unlink(instance)
 }
 
+func (s service) matchRequiredParameters(stackParams []model.StackRequiredParameter, instanceParams []model.InstanceRequiredParameter) error {
+	stackRequiredParameterNames := make([]string, len(stackParams))
+
+	for i, param := range stackParams {
+		stackRequiredParameterNames[i] = param.Name
+	}
+
+	for _, param := range instanceParams {
+		if !slices.Contains(stackRequiredParameterNames, param.StackRequiredParameterID) {
```

Comment:
I would go for a set of names which allows for easy and fast lookup. Right now we are looping over the params for every param (O(N^2)). It might not matter for the amount of parameters we have though.

Reply:
`paramNames := make(map[string]struct{})` is the way (using less memory than `bool`)


`paramNames[param.Name] = struct{}{}` (https://www.scaler.com/topics/golang/golang-set/)